### ISSUE TITLE
INFO:unityagents:Cannot use observations with imitation learning

### ISSUE THREAD

**Issue by RossMelbourne** [Unity-Technologies/ml-agents#537]
I received the above message when I tried using Visual Observations with an imitation learning scenario. I understand the overhead of Visual Observations but are there plans to make imitation learning work with Visual Observations?

**Comment by RossMelbourne** [comment]
It does work!  I set the Decision Frequency on each agent to 10 hoping that it would speed up the play during training and it did significantly.  The camera attached to each agent is only 160 x 120 but it appears large enough to do imitation learning.  I need to play with setting the FPS (not just -1) to see what impact it has on the training.

**Comment by Peter-Kocsis** [comment]
Hello, I'm facing with the same problem now. Could you help me, please, how did you use visual observations with imitation learning?

**Comment by RossMelbourne** [comment]
I am restating my comment from #540 here.  I was able to make 'visual observations only' work but did have to pass some dummy state data because I received a crash without it.  In addition to dummy state data, I was able to get imitation learning to work by using a very small visual observation width and height (120 x 160) and also skipping 10 frames before doing any learning (Decision Frequency on Agent set to 10). I set Quality Level to 1.  Also, I ignored the message 'Cannot use observations with imitation learning' after examining the python code and suspecting it would work.  I am assuming it is not supported because sending visual observations to python is cpu intensive but I am not sure if that is where the bottle neck might be.
